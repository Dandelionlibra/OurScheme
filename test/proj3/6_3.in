1
(define (apply-twice f x) (f (f x)))
(apply-twice (lambda (z) (if #f 1)) 1)

(let ((inc-or-nil (lambda (x) (cond ((< 5 3) 4)  )      )))
  ((lambda (h) (h 7)) inc-or-nil))

(define (apply-twice f x) (f (f x)))
(apply-twice (lambda (z)  (cond ((< 5 3) 4)  ) ) 1)

(define (apply-f f x) (f x))
(apply-f (lambda (y)  (cond ((< 5 3) 4)  )) 1)

(let ((inc-or-nil (lambda (x) (cond #f 1))))
  ((lambda (h) (h 7)) inc-or-nil))

(define (apply-twice f x) (f (f x)))
(apply-twice (lambda (z) (cond #f)) 1)

(define (apply-f f x) (f x))
(apply-f (lambda (y) (cond #f 1)) 1)



(let ((inc-or-nil (lambda (x) (if #f 1))))
  ((lambda (h) (h 7)) inc-or-nil))

(define (apply-twice f x) (f (f x)))
(apply-twice (lambda (z) (if #f)) 1)

(define (apply-f f x) (f x))
(apply-f (lambda (y) (if #f 1)) 1)

(let ((inc-or-nil (lambda (x) (if #f 1))))
  ((lambda (h) (h 7)) inc-or-nil))


(define (apply-f f x) (f x))
(apply-f (lambda (y) (if #f 1)) 1)
(apply-f 4 1)
(apply-f (lambda (y) (if #f 1)) 1 )


(define (foo n)
  (cond (
     (if #f 1) 1 (foo (- n 1))
  ) )
)

) ;
