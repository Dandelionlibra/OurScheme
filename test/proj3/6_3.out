Welcome to OurScheme!

> apply-twice defined

> ERROR (unbound parameter) : ( f
  x
)

> ERROR (no return value) : ( let
  ( ( inc-or-nil
      ( lambda
        ( x
        )
        ( cond
          ( ( <
              5
              3
            )
            4
          )
        )
      )
    )
  )
  ( ( lambda
      ( h
      )
      ( h
        7
      )
    )
    inc-or-nil
  )
)

> apply-twice defined

> ERROR (unbound parameter) : ( f
  x
)

> apply-f defined

> ERROR (no return value) : ( apply-f
  ( lambda
    ( y
    )
    ( cond
      ( ( <
          5
          3
        )
        4
      )
    )
  )
  1
)

> ERROR (COND format) : ( cond
  nil
  1
)

> apply-twice defined

> ERROR (COND format) : ( cond
  nil
)

> apply-f defined

> ERROR (COND format) : ( cond
  nil
  1
)

> ERROR (no return value) : ( let
  ( ( inc-or-nil
      ( lambda
        ( x
        )
        ( if
          nil
          1
        )
      )
    )
  )
  ( ( lambda
      ( h
      )
      ( h
        7
      )
    )
    inc-or-nil
  )
)

> apply-twice defined

> ERROR (incorrect number of arguments) : if

> apply-f defined

> ERROR (no return value) : ( apply-f
  ( lambda
    ( y
    )
    ( if
      nil
      1
    )
  )
  1
)

> ERROR (no return value) : ( let
  ( ( inc-or-nil
      ( lambda
        ( x
        )
        ( if
          nil
          1
        )
      )
    )
  )
  ( ( lambda
      ( h
      )
      ( h
        7
      )
    )
    inc-or-nil
  )
)

> apply-f defined

> ERROR (no return value) : ( apply-f
  ( lambda
    ( y
    )
    ( if
      nil
      1
    )
  )
  1
)

> ERROR (attempt to apply non-function) : 4

> ERROR (no return value) : ( apply-f
  ( lambda
    ( y
    )
    ( if
      nil
      1
    )
  )
  1
)

> foo defined

> ERROR (unexpected token) : atom or '(' expected when token at Line 2 Column 1 is >>)<<

> ERROR (no more input) : END-OF-FILE encountered
Thanks for using OurScheme!
